### PR TITLE
DEPS: Bump PyArrow to 6.0

### LINKED ISSUE #48014: Update pyarrow dependency from 1.0.1 to 3.0
Pandas 1.4 currently requires pyarrow 1.0.1 (released August 2020).
This issue is about discussing an update to the required pyarrow version, as suggested in #47781.

https://github.com/pandas-dev/pandas/pull/47781 implements a performance improvement that would require pyarrow 3.0 (released January 2021). 

Pyarrow releases now move pretty fast, with new releases coming out approx. every 3 months that add major functionality. As such, efforts such as arrow-backed storage would probably also gain from regularly updating the pyarrow dependency in pandas (as has been done in previous versions).

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #48014 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Would help https://github.com/pandas-dev/pandas/pull/47781 and https://github.com/pandas-dev/pandas/pull/49078

1.0.1 was released on August 21, 2020 and 5.0.0 was released July 28 2021 (almost a year later).

Last time optional dependencies were bumped (https://github.com/pandas-dev/pandas/pull/47048), the versions were based on the version released ~1 year _prior_, so given that logic, could even bump to 6.0.0 (October 26, 2021)

### CODE REVIEW

File: pandas/_testing/__init__.py [hunk 1]
```diff
@@ -195,7 +195,7 @@
     ]
 ]
 
-if not pa_version_under1p01:
+if not pa_version_under5p0:
```
Comment (phofl) [inline pandas/_testing/__init__.py:198]: This confuses me a bit. Does this mean that pyarrow is not installed at all? If minimum is 5, we can’t have anything below it

Comment (mroeschke) [reply pandas/_testing/__init__.py:198]: I think based on `pandas/compat/pyarrow.py`, the `not pa_version_under_x` means pyarrow is not installed or pyarrow is above version `x`.

So for these lines these objects are not created if pyarrow is not installed or pyarrow e.g. version 3.

Comment (phofl) [reply pandas/_testing/__init__.py:198]: Thx, was just confused about the under since that shouldn’t be a valid option anymore

**Comment by jbrockmendel** [comment]
> could even bump to 6.0.0

Go for it.  For a major release I think we're OK being really aggressive on this front.

**Comment by WillAyd** [comment]
I would also be +1 to a 6.0 bump

**Comment by jreback** [comment]
+1 in 6.0 as well

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @mroeschke

**Pull request merged**